torchbenchmark/models/hf_T5/__init__.py

```diff
@@ -0,0 +1,56 @@
+import torch
+import torch.optim as optim
+import torchvision.models as models
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import NLP
+from transformers import T5Tokenizer, T5ForConditionalGeneration
+from datasets import load_dataset
+
+class Model(BenchmarkModel):
+    task = NLP.LANGUAGE_MODELING
+
+    def __init__(self, device=None, jit=False):
+        super().__init__()
+        self.device = device
+        self.jit = jit
+
+        batch_size = 8
+        seq_length = 512
+
+        input_ids = torch.randint(0, 5000, (batch_size, seq_length)).to(device)
+        decoder_ids = torch.randint(0, 5000, (batch_size, seq_length)).to(device)
+
+        self.model = T5ForConditionalGeneration.from_pretrained('t5-small').to(device)
```

Comment:
I don't know how small is `t5-small` we could consider checking in the pth file to pin it.

Reply:
It's a couple hundred megabytes I think.

Reply:
You can see the size of each model by going to its files tab on the hub:
https://huggingface.co/t5-small/tree/main

so 230MB for t5-small

Reply:
wait so this 230MB is downloaded on setup? or in the repo? either way it sounds non-ideal.  do we need the real parameters?

Reply:
I've since changed it to not use the real parameters.

Reply:
If you just want to get the config, it's `T5Config.from_pretrained("t5-small")` and then you can tweak it and then create a model, but not pretrained, that is:
```
config = T5Config.from_pretrained("t5-small")
# edit config if need be
# now create a new untrained model
model = T5ForConditionalGeneration(config)
```

You can use `AutoConfig` and `AutoModel` everywhere and it'll automatically figure out the right class to use internally.

Reply:
so no download will happen.

and also if you run on the same machine the models are cached.